### DISCUSSION TITLE
Resolve build error while importing jquery in fomantic-ui.js with configured eslint rules

### DISCUSSION THREAD

**Discussion by semanticrelated** [fomantic/Fomantic-UI#1772] [Collections (form, grid, menu, message, table, ...)] [answered]
Hi Experts,

We are using webpack to build the front end.

In fomantic-ui.js, we are importing jquery in the first line itself.

// jQuery
//var jQuery = require("jquery");
import jQuery = require('jquery');

Package.json has the below dependencies added

"fomantic-ui-less": "^2.8.3"

But during the build it fails, with below errors. Any thoughts/pointers/reference code on how to resolve this will be really helpful.

[INFO] ERROR in ./src/main/webpack/fomantic-ui.js 3:14
[ERROR] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[INFO] Module parse failed: Unexpected token (3:14)
[ERROR]
[INFO] File was processed with these loaders:
[ERROR] npm ERR! A complete log of this run can be found in:
[INFO]  * ./node_modules/eslint-loader/dist/cjs.js
[ERROR] npm ERR!     C:\Users\<username>\AppData\Roaming\npm-cache\_logs\2020-11-17T16_02_31_276Z-debug.log
[INFO] You may need an additional loader to handle the result of these loaders.
[INFO] | // jQuery
[INFO] | //var jQuery = require("jquery");
[INFO] > import jQuery = require('jquery');
[INFO] |
[INFO] | /* Import Fomantic UI JS from Definitions folder */
[INFO]  @ ./src/main/webpack/index.js 5:0-23

**Comment by lubber-de** [comment, marked as answer]
We had a similiar discussion using webpack before. Maybe https://github.com/fomantic/Fomantic-UI/discussions/1591#discussioncomment-92902 helps?

**Reply by semanticrelated** [reply]
Hello,

Thank you for your reply.

Did go through the complete details you shared. In fomantic-ui.js we are importing jquery first, followed by the fomantic-ui related stuffs.

However, due to some eslint rules, getting the error[mentioned above] for the first line in the fomantic-ui.js

Is there any other way we can import the jquery or am I doing something wrong.

**Reply by lubber-de** [reply]
What is the "first line of fomantic-ui.js" and what is eslint complaining about? "Unexpected token (3:14)" does not help, if we don't know whats at [3:14] 😉 

Please share the relevant code /token that eslint/webpack is complaining about

Otherwise maybe the webpack or eslint people can help better

**Reply by semanticrelated** [reply]
First line in fomantic-ui.js corresponds to the below :

// jQuery
//var jQuery = require("jquery");
import jQuery = require('jquery');

**Reply by lubber-de** [reply]
Well, these are actually 3 lines 😉
According to "3:14" the error is the `=` (which is line 3, character 14)
Import together with require assignment is invalid syntax. See https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Statements/import
Whatever your eslint is doing... 

As I mentioned in the above thread try the approach 

```javascript 
import {$,jQuery} from 'jquery';
```
instead
